recognize.py

```diff
@@ -1,37 +1,114 @@
 import argparse
 import pickle
 import numpy as np
 import cv2
+import time
+import imutils
+import datetime
 from PIL import Image
 from identify_faces import load_image_file, identify_face
 from extract_embeddings import extract_with_dlib
 
+def get_greetings_text(name):
+  now = datetime.datetime.now()
+  if now.time() < datetime.time(12):
+    return "Bom dia {}!".format(name)
+  if now.time() < datetime.time(18):
+    return "Boa tarde {}!".format(name)
+  return "Boa noite {}!".format(name)
+
+def use_image(image_path):
+  test_image = load_image_file(image_path)
+  recognizer = pickle.loads(open('output/recognizer.pickle', "rb").read())
+  le = pickle.loads(open('output/le.pickle', "rb").read())
+
+  face_locations = identify_face(test_image)
+  no = len(face_locations)
+
+  test_image_encodings = extract_with_dlib(test_image, face_locations)
+  for i in range(no):
+    test_image_encoding = test_image_encodings[i]
+    class_probabilities = recognizer.predict_proba([test_image_encoding])
+    j = np.argmax(class_probabilities[0])
+    proba = class_probabilities[0][j]
+    name = le.classes_[j]
+    top, right, bottom, left = face_locations[i]
+    text = "{}: {:.2f}%".format(name, proba * 100)
+    y = bottom - 10 if bottom - 10 > 10 else bottom + 10
+    cv2.putText(test_image, text, (left, y), cv2.FONT_HERSHEY_SIMPLEX, 0.45, (0, 0, 255), 2)
+    cv2.rectangle(test_image, (left, bottom), (right, top), (0, 0, 255), 2)
+
+  pil_image = Image.fromarray(test_image)
+  pil_image.resize((100,100))
+  pil_image.show()
+
+def use_video():
+  recognizer = pickle.loads(open('output/recognizer.pickle', "rb").read())
+  le = pickle.loads(open('output/le.pickle', "rb").read())
+
+  print("[INFO] starting video stream...")
+  vs = VideoStream(src=0).start()
+  time.sleep(2.0)
+
+  fps = FPS().start()
+
+  while True:
+    frame = vs.read()
+
+    frame = imutils.resize(frame, width=600)
+    (h, w) = frame.shape[:2]
+
+    recognizer = pickle.loads(open('output/recognizer.pickle', "rb").read())
+    le = pickle.loads(open('output/le.pickle', "rb").read())
+
+    face_locations = identify_face(frame)
+    no = len(face_locations)
+
+    test_image_encodings = extract_with_dlib(frame, face_locations)
+    for i in range(no):
+      test_image_encoding = test_image_encodings[i]
+      class_probabilities = recognizer.predict_proba([test_image_encoding])
+      j = np.argmax(class_probabilities[0])
+      proba = class_probabilities[0][j]
+      name = le.classes_[j]
+      top, right, bottom, left = face_locations[i]
+      text = "{}: {:.2f}%".format(name, proba * 100)
+      greetings = get_greetings_text(name.capitalize())
+      y = bottom - 10 if bottom - 10 > 10 else bottom + 10
+      cv2.putText(frame, greetings, (10, (i + 1) * 35), cv2.FONT_HERSHEY_SIMPLEX, 1.20, (0, 0, 255), 2)
+      cv2.putText(frame, text, (left, y), cv2.FONT_HERSHEY_SIMPLEX, 0.45, (0, 0, 255), 2)
+      cv2.rectangle(frame, (left, bottom), (right, top), (0, 0, 255), 2)
```

Comment:
Is there any reason why we should not put this for loop in a separate function as it is repeated on both use_videoand use_image functions?

Reply:
Good point Dani. I moved the "for" code to a shared function.